test/functional/services/apps_menu.ts

```diff
@@ -116,17 +117,50 @@ export class AppsMenuService extends FtrService {
       category,
     }: { closeCollapsibleNav?: boolean; category?: string } = {}
   ) {
+    await this.waitUntilLoadingHasFinished();
+
     try {
       this.log.debug(`click "${name}" app link`);
       await this.openCollapsibleNav();
       let nav;
       if (typeof category === 'string') {
+        // we can search within a specific section of the side nav
         nav = await this.testSubjects.find(`collapsibleNavGroup-${category}`);
+        const link = await nav.findByPartialLinkText(name);
+        await link.click();
       } else {
-        nav = await this.testSubjects.find('collapsibleNav');
+        // we need to search our app link in the whole side nav
+        // first, we get all the links, along with their inner text
+        const allLinks = await this.testSubjects.findAll('collapsibleNavAppLink');
+        const allLinksTexts = await Promise.all(
+          allLinks.map(
+            (link: WebElementWrapper): Promise<{ link: WebElementWrapper; text: string }> =>
+              new Promise((resolve) => {
+                link.getVisibleText().then((text: string) => resolve({ link, text }));
+              })
+          )
+        );
```

Comment:
I don't how many links in the particular case, but if ~10+ I would consider using cheerio to parse all the links inside the navigation element:
```
const $ = await nav.parseDomContent();
$(<link locator>).attr(..)
```

Might be not a great idea if we rely on visible text, but pretty simple if it is an attribute

Reply:
As discussed, I can't use `cheerio` in this particular case, as I intend to `click()` in the link that has a matching _App_ text.